pandas/io/packers.py

```diff
@@ -214,6 +217,14 @@ def convert(values):
     if dtype == np.object_:
         return v.tolist()
 
+    if is_categorical_dtype(dtype):
+        return {
```

Comment:
so ignore my first comment here. move this logic to ``encode/decode``. then all will work.